account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py

```diff
@@ -29,11 +29,37 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff,
                 rec.debit_move_id.move_id if
                 rec.credit_move_id.journal_id.type == 'purchase' else
                 rec.credit_move_id)
+            lines = []
+            analytic_accounts = []
+
+            # We get the analytic account if this exists in the invoice
+            for line in invoice_move.move_id.line_ids.filtered(
+                    lambda r: r.analytic_account_id):
+                analytic_accounts.append(line.analytic_account_id.id)
+            total_analytic_account = len(analytic_accounts)
```

Comment:
you can use `if analytic_accounts:` 

You don't need to use len.

if you test an empty list you will get False.